src/memory.ml

```diff
@@ -22,18 +22,18 @@ let set_memory wasm_mod initial maximum export_name (segments : segment list)
     shared =
   let split_segments segments =
     List.fold_right
-      (fun { name; passive; offset; size }
+      (fun { data; passive; offset; size }
            (segment_names, segment_passive, segment_offsets, segment_sizes) ->
```

Comment:
Could we rename `segment_names` here too? Otherwise looks great!

Reply:
hater :P

Reply:
Updated